consensus/spos/worker.go

```diff
@@ -189,12 +193,38 @@ func checkNewWorkerParams(
 
 func (wrk *Worker) receivedSyncState(isNodeSynchronized bool) {
 	if isNodeSynchronized {
-		if len(wrk.consensusStateChangedChannel) == 0 {
-			wrk.consensusStateChangedChannel <- true
+		select {
+		case wrk.consensusStateChangedChannel <- true:
+		default:
 		}
 	}
 }
 
+func (wrk *Worker) ReceivedHeader(headerHandler data.HeaderHandler, headerHash []byte) {
+	if headerHandler.GetShardID() != wrk.shardCoordinator.SelfId() ||
+		int64(headerHandler.GetRound()) != wrk.rounder.Index() {
+		return
+	}
+
+	wrk.mutReceivedHeadersHandler.RLock()
+	for _, handler := range wrk.receivedHeadersHandlers {
+		handler(headerHandler)
+	}
+	wrk.mutReceivedHeadersHandler.RUnlock()
+
+	select {
+	case wrk.consensusStateChangedChannel <- true:
+	default:
```

Comment:
time out wait here ?

Reply:
The channel is buffered with size=1. Actually the behavior of this code is: just to signal that an event happened. If the channel is not empty it is ok, as the notification would be send anyway so I do not need to stay here anymore